src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    const stateCopy = { ...currentState };
+
+    if (item.type === 'addProperties') {
+      Object.assign(stateCopy, { ...item.extraData });
+    }
+
+    if (item.type === 'removeProperties') {
+      for (const el of item.keysToRemove) {
+        delete stateCopy[el];
+      }
+    }
+
+    if (item.type === 'clear') {
+      for (const param in stateCopy) {
+        delete stateCopy[param];
+      }
+    }
```

Comment:
Instead of using multiple if statements, consider refactoring the code to use a switch statement for handling different action types. A switch statement is more appropriate when you have a limited number of conditions to check, and it will also improve the readability of the code.

Reply:
ok